### ISSUE TITLE
Breaking change with inserting data into "json" type fields

### ISSUE THREAD

**Issue by trival** [hasura/graphql-engine#3642]
I recently upgraded hasura from beta-9 to stable 1.0.0. 
My tests indicate that inserting into a field of type "json" now stringifies the data automatically, whereas before i had to stringify the data by hand. Now inserting the string "[1, 2, 3]“ returns the value "\"[1, 2, 3]\"" in queries, whereas before it was just "[1, 2, 3]". 
This is not the case with updating. Here the automatic stringification does not happens. 
Is this expected behavior, and will it stay in future versions?

**Comment by rakeshkky** [comment]
Hi @trival,

I tried the following to reproduce the issue with `v1.0.0` :

Postgres schema
```sql
create table test (id serial primary key, json_col json not null)
```
Test queries
1.
```graphql
mutation m {
  insert_test(objects: [{json_col: "[1,2,3]"}]){
    affected_rows
    returning{
      id
      json_col
    }
  }
}
```
Response
```json
{
  "data": {
    "insert_test": {
      "affected_rows": 1,
      "returning": [
        {
          "id": 1,
          "json_col": "[1,2,3]"
        }
      ]
    }
  }
}
```

2.
```graphql
query a {
  test{
    id
    json_col
  }
}
```
Response
```json
{
  "data": {
    "test": [
      {
        "id": 1,
        "json_col": "[1,2,3]"
      }
    ]
  }
}
```

The behaviour of the above queries seems to be fine to me.

Please give GraphQL queries you tried to reproduce the issue. Thank you.

**Comment by trival** [comment]
Hi @rakeshkky 

Thank you for the quick reply. I can confirm your result. But when I add a relation, the stringification starts: 

```
mutation {
  insert_test(objects: [{json_col: "[1, 2, 3]", test2: {data: {name: "test_name"}}}]) {
    returning {
      id
      json_col
    }
  }
}
```

response: 
```
{
  "data": {
    "insert_test": {
      "returning": [
        {
          "json_col": "\"[1, 2, 3]\"",
          "id": "bd5c9f05-0ff7-4f8d-b5c5-a633436513e5"
        }
      ]
    }
  }
}
```

Postgres schema (all created with hasura console)
```
CREATE TABLE public.test
(
    id uuid NOT NULL DEFAULT gen_random_uuid(),
    json_col json,
    test2_relation uuid,
    CONSTRAINT test_pkey PRIMARY KEY (id),
    CONSTRAINT test_test2_relation_fkey FOREIGN KEY (test2_relation)
        REFERENCES public.test2 (id) MATCH SIMPLE
        ON UPDATE CASCADE
        ON DELETE CASCADE
)
CREATE TABLE public.test2
(
    id uuid NOT NULL DEFAULT gen_random_uuid(),
    name text COLLATE pg_catalog."default",
    CONSTRAINT test2_pkey PRIMARY KEY (id)
)
```

**Comment by trival** [comment]
Postgres Version (from postgres docker image): 
PostgreSQL 11.4 (Debian 11.4-1.pgdg90+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516, 64-bit

**Comment by trival** [comment]
I also see that this only affects the insert mutation itself: 
```
query {
  test {
    json_col
  }
}
```
returns as expected
```
{
  "data": {
    "test": [
      {
        "json_col": "[1, 2, 3]"
      },
    ]
  }
}
```

**Comment by trival** [comment]
Thanks for taking care of this!
Btw: I am using autogenerated Typescript types, that required the "json" type field to be a string in my code. That is why I was always applying `JSON.stringify` before inserting. By investigating the issue I learned that I can actually pass in the data as is, which makes more sense and is more usefull.

**Comment by 0x777** [comment]
This has been fixed in `v1.1.0-beta.3`.
